ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobPutHandler.java

```diff
@@ -202,6 +202,13 @@ private Callback<Void> securityPostProcessRequestCallback(BlobInfo blobInfo) {
           RetainingAsyncWritableChannel channel =
               new RetainingAsyncWritableChannel(frontendConfig.maxJsonRequestSizeBytes);
           restRequest.readInto(channel, fetchStitchRequestBodyCallback(channel, blobInfo));
+        } else if (RestUtils.isDatasetVersionStitchRequest(restRequest)) {
+          if (RestUtils.isDatasetVersionQueryEnabled(restRequest.getArgs())) {
+            addDatasetVersion(blobInfo.getBlobProperties(), restRequest);
+          }
+          RetainingAsyncWritableChannel channel =
```

Comment:
I suggest we just reuse the named blob stitch header so we can move the addDatasetVersion logic to the first if statement